scripts/Reader.py.experimental

```diff
@@ -81,49 +73,88 @@ class Mfrc522Reader(object):
     def cleanup():
         GPIO.cleanup()
 
-
 class Rdm6300Reader:
-    def __init__(self):
+    def __init__(self,param = None):
+        import serial
         device = '/dev/ttyS0'
         baudrate = 9600
         ser_timeout = 0.1
         self.last_card_id = ''
         try:
             self.rfid_serial = serial.Serial(device, baudrate, timeout=ser_timeout)
+            self.serial_SerialException = serial.SerialException
         except serial.SerialException as e:
             logger.error(e)
             exit(1)
+        
+        self.number_format = ''
+        if param is not None:
+            nf = param.get("numberformat")
+            if nf is not None:
+                self.number_format = nf
+        
+    def convert_to_weigand26_when_checksum_ok(self,raw_card_id):
+        weigand26 = []
+        xor = 0
+        for i in range(0, len(raw_card_id)>>1): 
+            val = int(raw_card_id[i*2:i*2+2],16)
+            if (i < 5):
+                xor = xor ^ val
+                weigand26.append(val)
+            else:
+                chk = val
+        if (chk == val):
+            return weigand26
+        else:
+            return None
 
     def readCard(self):
-        byte_card_id = b''
+        byte_card_id = bytearray()
 
         try:
             while True:
                 try:
-                    read_byte = self.rfid_serial.read()
-
-                    if read_byte == b'\x02':    # start byte
-                        while read_byte != b'\x03':     # end bye
-                            read_byte = self.rfid_serial.read()
-                            byte_card_id += read_byte
-
-                        card_id = byte_card_id.decode('utf-8')
-                        byte_card_id = ''
-                        card_id = ''.join(x for x in card_id if x in string.printable)
-
-                        # Only return UUIDs with correct length
-                        if len(card_id) == 12 and card_id != self.last_card_id:
-                            self.last_card_id = card_id
-                            self.rfid_serial.reset_input_buffer()
-                            return self.last_card_id
-
-                        else:   # wrong UUID length or already send that UUID last time
-                            self.rfid_serial.reset_input_buffer()
+                    wait_for_start_byte = True
+                    while True:
+                        read_byte = self.rfid_serial.read()
+
+                        if (wait_for_start_byte):
+                            if read_byte == b'\x02':
+                                wait_for_start_byte = False
+                        else:
+                            if read_byte != b'\x03':        #could get stuck here, check len? check timeout by len == 0??
+                                byte_card_id.extend(read_byte)
+                            else:
+                                wait_for_start_byte = True
+                                break
```

Comment:
Why do you set `wait_for_start_byte` to `True` again here if you leave the loop right after anyway?

Reply:
Yes, you are right, this is not needed here.
thanks for that finding